src/main/resources/templates/oidcRedirect.html

```diff
@@ -14,120 +14,257 @@
+            </div>
+            <div class="form-group">
+                <label for="redirect_uri">redirect_uri:</label>
+                <input type="text" id="redirect_uri" name="redirect_uri" th:value="${redirect_uri}"
+                    class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="response_mode">response_mode:</label>
+                <input type="text" id="response_mode" name="response_mode" th:value="${response_mode}"
+                    class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="response_type">response_type:</label>
+                <input type="text" id="response_type" name="response_type" th:value="${response_type}"
+                    class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="scope">scope:</label>
+                <input type="text" id="scope" name="scope" th:value="${scope}" class="form-control" />
+            </div>
+            <div class="form-group">
+                <label for="state">state:</label>
+                <input type="text" id="state" class="form-control" name="state" th:value="${state}" />
+            </div>
+            <input id="post_button" type="submit" value="Submit POST" class="btn btn-primary" disabled="disabled">
+        </form>
+    </div>
 
-<br/>
+    <br />
 
-<button class="accordion large-font">
-    Click to view the GET version of Step 2 of the LTI Core Standard Launch
-</button>
-<div class="panel">
-    <div class="alert alert-primary">GET URL: <span th:text="${oidcEndpointComplete}"/></div>
-</div>
-<a th:href="${oidcEndpointComplete}" class="btn btn-primary" role="button" aria-pressed="true">Perform GET as Step 2</a>
+    <button class="accordion large-font">
+        Click to view the GET version of Step 2 of the LTI Core Standard Launch
+    </button>
+    <div class="panel">
+        <div class="alert alert-primary">GET URL: <span th:text="${oidcEndpointComplete}" /></div>
+    </div>
+    <a th:href="${oidcEndpointComplete}"  id="get_button" class="btn btn-primary disabled" role="button" aria-pressed="true">Perform GET as Step
+        2</a>
+
+<script th:inline="javascript">
+
+    const nonceStateCheck = async () => {
+
+    // placeholder for state and nonce validations
+    let stateValid = false, nonceValid = false, stateNonceValid = false;
+
+    // state and nonce values. These should be generated on the server side
+    const stateHash = [[${ state }]];
+    const nonce = [[${ nonce }]];
+
+    // origin of the LTI launch. Currently, * is the only value the LMS appears to accept
+    let origin = [[${oidc_authorization_uri }]];
+    //let origin = '*';
+
+    // Create an instance of LTIPostMessage
+    let ltiPostMessage = new LTIPostMessage([[${ initiation_dto.ltiStorageTarget }]]);
+
+    const verifyStateAndNonce = () => {
```

Comment:
A lot of this code is duplicated on the nonceStateCheck.html page as well. Could we consolidate the duplicates into utils?

Reply:
I did not do that because we are using the thymeleaf replacement options: like : [[${ state }]], and they are slightly different (surely that can be managed with more parameters)... so it would imply to create the method with a lot of attributes...  feasible, but well... I will leave that to Phil if he wants to do it.  I'm not a javascript fan, and I know that I will spend a long time to make it work... :(